nemo/collections/asr/modules/rnnt.py

```diff
@@ -368,6 +368,20 @@ def batch_concat_states(self, batch_states: List[List[torch.Tensor]]) -> List[to
 
         return state_list
 
+    def batch_replace_states_mask(
+        self, src_states: list[torch.Tensor], dst_states: list[torch.Tensor], mask: torch.Tensor,
+    ):
+        """Replace states in dst_states with states from src_states using the mask"""
+        # same as `dst_states[0][mask] = src_states[0][mask]`, but non-blocking
+        torch.where(mask.unsqueeze(-1), src_states[0], dst_states[0], out=dst_states[0])
```

Comment:
Please state somehow that `batch_replace_states_mask` is an in-place method.

Reply:
Hm, can you please explain how to do this? This is described in the docstring.

Reply:
`def batch_replace_states_mask_(...)`
```
"""Replace states in dst_states with states from src_states using the mask.
In-place operation."""
```

Reply:
I think this style is not appropriate here: this is used for in-place modification when calling object methods, e.g. 

```python
x = torch.rand(...)
y = torch.rand(...)
x.logical_and_(y) # in-place
```

Since `batch_replace_states_mask` does not modify the **`self`** state, it should not be marked with underscore (it modifies the object passed, not self).

Cmp: `x.logical_and_(y)` (modify `x`) and `torch.logical_and(x, y, out=x)` (`torch` is not modified, but `x` is still modified)

Reply:
About `self`: why do we need `self` in the definition if nothing is modified/used?

Reply:
Hmm, maybe we speak about different things. 
I mean, that `batch_replace_states_mask` is not an in-place method (and should not be marked with underscore), since it does not modify the caller (self).

Reply:
> I mean, that batch_replace_states_mask is not an in-place method (and should not be marked with underscore), since it does not modify the caller (self).

I agree. My question is why do we need self here since it is not used whatsoever?

Reply:
Ah, do you mean to make the method static/classmethod?

Reply:
Yep

Reply:
Ah, ok, thanks! I will fix it)

Reply:
Done, made a classmethod.
Sorry for the misunderstanding!